qadence/backends/pulser/backend.py

```diff
@@ -95,18 +96,37 @@ def make_sequence(circ: QuantumCircuit, config: Configuration) -> Sequence:
 # TODO: make it parallelized
 # TODO: add execution on the cloud platform
 def simulate_sequence(
-    sequence: Sequence, config: Configuration, state: Tensor
-) -> SimulationResults:
-    simulation = QutipEmulator.from_sequence(
-        sequence,
-        sampling_rate=config.sampling_rate,
-        config=config.sim_config,
-        with_modulation=config.with_modulation,
-    )
-    if state is not None:
-        simulation.set_initial_state(qutip.Qobj(state.cpu().numpy()))
+    sequence: Sequence, config: Configuration, state: Tensor, n_shots: int | None = None
+) -> SimulationResults | Counter:
+    if config.cloud_credentials is not None and config.cloud_platform is not None:
+        client = get_client(config.cloud_credentials)
+        serialized_sequence = sequence.to_abstract_repr()
+        params: list[dict] = [{"runs": n_shots, "variables": {}}]
+
+        batch = client.create_batch(
+            serialized_sequence,
+            jobs=params,
+            emulator=str(config.cloud_platform),
+            wait=True,
+        )
+        job = list(batch.jobs.values())[0]
+        return Counter(job.result)
 
-    return simulation.run(nsteps=config.n_steps_solv, method=config.method_solv)
+    else:
+        simulation = QutipEmulator.from_sequence(
+            sequence,
+            sampling_rate=config.sampling_rate,
+            config=config.sim_config,
+            with_modulation=config.with_modulation,
+        )
+        if state is not None:
+            simulation.set_initial_state(qutip.Qobj(state.cpu().numpy()))
+
+        sim_result = simulation.run(nsteps=config.n_steps_solv, method=config.method_solv)
+        if n_shots is not None:
+            return sim_result.sample_final_state(n_shots)
```

Comment:
Is this returning a `Counter` instance as stated in the return type?

Reply:
Yes, it is